### ISSUE TITLE
Use pointer cursor instead of hand cursor when hovering over button

### ISSUE THREAD

**Issue by Shomnipotence** [microsoft/vscode#157430]
**- What it looks now**

normal cursor
![image](https://user-images.githubusercontent.com/6630660/183289986-d000169e-5eb5-4f99-a486-83838fc7a8f4.png)

Hand Cursor when hovering over the button
![image](https://user-images.githubusercontent.com/6630660/183289927-41e5c283-cd2e-42e4-aaaf-9dc7d546ec10.png)

Although we know that these buttons are essentially hyperlinks, we still want to erase this distinction.It should use a normal pointer cursor in the button like apps developed with other frameworks.

Including
Sidebar Item
![image](https://user-images.githubusercontent.com/6630660/183443601-cf934f64-938e-4300-9419-3f865b6b70e1.png)

Menu Item
![image](https://user-images.githubusercontent.com/6630660/183319880-1a17f024-ed1b-465e-bac6-3699bb8a1e25.png)

Tabs

Explorer

Statusbar Item

**- What it should be**

When hovering over the button, it should be a normal cursor, like in other apps.

For example, setting the left navigation item of the app
![image](https://user-images.githubusercontent.com/6630660/183290127-8d256e23-ecb5-4fd7-b28f-c12a75ffdccd.png)

Most of the itmes have Highlight , so the hand pointer is uesless
All the apps built by WinForms,WPF,UWP or WinUI3,Qt or other frames did not have this bug(may you think that it is a feature)

**Comment by gjsjohnmurray** [comment]
I think the hand cursor is intended as a hint that you can drag the activity bar icons, for example to reorder them or move them to the side panel.

**Comment by Shomnipotence** [comment]
> I think the hand cursor is intended as a hint that you can drag the activity bar icons, for example to reorder them or move them to the side panel.

What is the reason of hand cursor on Menu Highlight Item？
![image](https://user-images.githubusercontent.com/6630660/183319761-9646d51a-7c44-4ba1-889b-5ede002cb398.png)

**Comment by isidorn** [comment]
I somewhat agree with this but let's keep this issue open as a feature request to gather more feedback
fyi @sandy081

**Comment by gjsjohnmurray** [comment]
By changing the pointer it's more obvious when the pointer is over an actionable menu item. When over a disabled one the pointer remains an arrow. This is consistent with what happens when pointing at action buttons in the UI.

**Comment by Shomnipotence** [comment]
> By changing the pointer it's more obvious when the pointer is over an actionable menu item. When over a disabled one the pointer remains an arrow. This is consistent with what happens when pointing at action buttons in the UI.

Most of the itmes have Highlight Status , so the hand pointer is uesless

**Comment by gjsjohnmurray** [comment]
VS Code's behaviour in this area is consistent with how the GitHub web UI behaves for me.

**Comment by sandy081** [comment]
Agreed with @gjsjohnmurray. Its consistent now across VS Code to use hand cursor for all items those are clickable. 

CC @misolori

**Comment by Shomnipotence** [comment]
@misolori 
I don't think all the clickable items should use hand cursor.

It's inconsistent with Windows' default behavior, buttons are not hyperlinks, this is an app not a website.

**Comment by gjsjohnmurray** [comment]
> this is an app not a website

https://code.visualstudio.com/blogs/2021/10/20/vscode-dev

**Comment by Shomnipotence** [comment]
> > this is an app not a website
> 
> https://code.visualstudio.com/blogs/2021/10/20/vscode-dev

Yes I know. But Outlook is a website, too. And the Office Online.

**Comment by misolori** [comment]
In the past, we've discussed whether we should drop the hand cursor and the team consensus was that it's still needed for the web but on desktop, we need to ensure we still have hover feedback as an alternate. I can bring this up again in our UX Sync.

**Comment by Tyriar** [comment]
Maybe we should change it only for context menus and the top menu? The pointer elsewhere in the UI is a good thing imo and a big change.

**Comment by Shomnipotence** [comment]
> Maybe we should change it only for context menus and the top menu? The pointer elsewhere in the UI is a good thing imo and a big change.

the explorer has highlight，so it does not need hand cursor

![image](https://user-images.githubusercontent.com/6630660/183535434-0dbd0fa3-c418-4f46-801d-89cc86980bae.png)

**Comment by RayGicEFL** [comment]
I think it is better to do something else like add GUI for `launch.json` instead of arguing on this trivial issue.

**Comment by Shomnipotence** [comment]
> I think it is better to do something else like add GUI for `launch.json` instead of arguing on this trivial issue.

It is based on Web so you can change the appearance by editing *.css

**Comment by Shomnipotence** [comment]
> I think it is better to do something else like add GUI for `launch.json` instead of arguing on this trivial issue.

I am agree with “GUI for launch.json”, but I think the cursor is important,too.

**Comment by Hardiol** [comment]
I agree with you

**Comment by misolori** [comment]
Brought this up in our ux sync today and the consensus was:

- We should explore this in the next iteration, early on
- Revisit areas that don't provide hover feedback
- Show underlines for blue links (since the pointer was the main feedback)
- Read the OS guidelines for mac/win/linux to see if there are guidance
- Possibly keep things consistent across platforms

So we'll start exploring this in our next iteration.

**Comment by Shomnipotence** [comment]
> 今天在我们的用户体验同步中提出了这一点，共识是：
> 
> * 我们应该在下一次迭代中尽早探索这个
> * 重新访问不提供悬停反馈的区域
> * 显示蓝色链接的下划线（因为指针是主要反馈）
> * 阅读 mac/win/linux 的操作系统指南，看看是否有指南
> * 可能保持跨平台的一致性
> 
> 因此，我们将在下一次迭代中开始探索这一点。这是一个在[misolori/promising-mammal-pointer中工作的演示](https://github.com/microsoft/vscode/tree/misolori/promising-mammal-pointer)
> 
>  CleanShot.2022-08-10.at.08.48.45.mp4

This is a great start!

But in this earliest version I need to point out (though I know this is just an early demo of yours):

In macOS, the search box does not have a hover highlight effect.
On Windows and macOS, there is no hover change for normal buttons (there used to be a Reveal effect during the Windows 10 Creators Update)

![_reveal-highlight-fluent-design-demo](https://user-images.githubusercontent.com/6630660/184048100-a2db3f1e-9a2d-40b2-b65d-873bccee26a6.gif)

**Comment by isidorn** [comment]
Assigning this back to the backlog since we will not have time in September to think about this.
Also unassigning myself so design ca drive this. I personally think both options are fine.

**Comment by Shomnipotence** [comment]
> Assigning this back to the backlog since we will not have time in September to think about this. Also unassigning myself so design ca drive this. I personally think both options are fine.

I think the design study is over based on @miguelsolorio 's feedback, the conclusion is clear, we should use the default cursor instead of the hand cursor in the hover effect on the desktop

**Comment by Shomnipotence** [comment]
> Brought this up in our ux sync today and the consensus was:
> 
> * We should explore this in the next iteration, early on
> * Revisit areas that don't provide hover feedback
> * Show underlines for blue links (since the pointer was the main feedback)
> * Read the OS guidelines for mac/win/linux to see if there are guidance
> * Possibly keep things consistent across platforms
> 
> So we'll start exploring this in our next iteration. Here's a demo of it working in [misolori/promising-mammal-pointer](https://github.com/microsoft/vscode/tree/misolori/promising-mammal-pointer)
> 
>  CleanShot.2022-08-10.at.08.48.45.mp4

I don't know why the assignment was removed, I think the conclusion of the design study is clear that this is a change that must be made.

**Comment by gjsjohnmurray** [comment]
> I don't know why the assignment was removed

@Shomnipotence it has been reassigned to @daviddossett because @miguelsolorio has left the team.

**Comment by Shomnipotence** [comment]
> > I don't know why the assignment was removed
> 
> @Shomnipotence it has been reassigned to @daviddossett because @miguelsolorio has left the team.

I don't think it's necessary, the design research is over, the conclusions have been made, and it's time for the developers to perform.

**Comment by Ryutaku** [comment]
After paying attention to this topic for a long time, what was the final conclusion regarding this topic? When will the cursor transition from the hand cursor back to the normal pointer cursor  when hovering over tab pages, the left toolbar, and the menu bar?

**Comment by Ryutaku** [comment]
When the mouse hovers over the menu bar, left toolbar, and tab pages, the cursor displayed as a hand shape makes me very uncomfortable and unfamiliar.

**Comment by crizzo-xerox** [comment]
The issue I have with the hand cursor is it blocks the tooltip and I have trouble reading it.

**Comment by kachkaev** [comment]
If I understand correctly, there is a simple rule for the web. If it’s a link with an href, it’s a hand cursor. If it’s an interactive element, the cursor is a pointer, a drag hand, resize arrows etc. Most on-page elements have a hover state which communicates their interactivity.

Can I ctrl+click it to open the result in a new tab? Can I right click and copy the URL?
- **yes** → Hand
- **no** → Something else (default pointer, drag hand, resize arrows, question mark, etc)

Most elements in VS Code cannot be opened in a new tab, so the cursor should not change to a hand.

**Comment by westerlind** [comment]
This is of course just my opinion, but I think that as VS Code is a desktop app, it should behave like a desktop app. Meaning that it does not matter how web sites normally works.

The standard in desktop apps for all the big operating systems is that there is only a hand cursor when hovering something that is intended to be understood as a link, and that means that it is almost always something that does not look like a button, but is just text that might have a different color and/or be underlined to indicate that it's a link, and it is something that is expected to most likely open a web page, or at least lead to "somewhere else", possibly opening a different app, or in some cases open a folder in the file browser of the OS for example.

I don't know why this should be any different in VS Code. The behaviour that everything clickable is treated as if it was a link is just confusing and feels out of place, and doesn't help the user to understand the behaviour of the application either, as it's different from almost all other applications they will be using.

Someone could of course say that everyone is very used to how the web works, and that this might make sense because of that. But even while browsing the web you will not in any web browser I have seen get a hand cursor while hovering the tabs or the buttons in the toolbar, or menu items. The web browser itself distinguishes between what is a desktop app and what is a web site. So it still doesn't make any sense from the perspective of being familiar to the user.

You can of course run it as a web app too these days, and I guess that there might be more arguments for it to behave like a web site then, but it's still primarily a desktop app and a desktop app should behave like a desktop app to avoid creating unnecessary friction for the user.

**Comment by martinszeltins** [comment]
Is there a setting or extension I can use to change the mouse cursor?

**Comment by scleikas** [comment]
> This is of course just my opinion, but I think that as VS Code is a desktop app, it should behave like a desktop app. Meaning that it does not matter how web sites normally works.
> 
> The standard in desktop apps for all the big operating systems is that there is only a hand cursor when hovering something that is intended to be understood as a link, and that means that it is almost always something that does not look like a button, but is just text that might have a different color and/or be underlined to indicate that it's a link, and it is something that is expected to most likely open a web page, or at least lead to "somewhere else", possibly opening a different app, or in some cases open a folder in the file browser of the OS for example.

Agree with this 100%.

**Comment by jarnoleikas** [comment]
> This is of course just my opinion, but I think that as VS Code is a desktop app, it should behave like a desktop app. Meaning that it does not matter how web sites normally works.
> 
> The standard in desktop apps for all the big operating systems is that there is only a hand cursor when hovering something that is intended to be understood as a link, and that means that it is almost always something that does not look like a button, but is just text that might have a different color and/or be underlined to indicate that it's a link, and it is something that is expected to most likely open a web page, or at least lead to "somewhere else", possibly opening a different app, or in some cases open a folder in the file browser of the OS for example.
> 
> I don't know why this should be any different in VS Code. The behaviour that everything clickable is treated as if it was a link is just confusing and feels out of place, and doesn't help the user to understand the behaviour of the application either, as it's different from almost all other applications they will be using.

Agree with this 100%.

**Comment by Ryutaku** [comment]
This is my solution：https://github.com/Ryutaku/vscode-default-cursor

**Comment by Tyriar** [comment]
Realistically this isn't going to happen, it's too big/visible of a change and no one is passionate about it within the team to champion it. We close off features that aren't likely going to happen within the next 24 months https://github.com/Microsoft/vscode/wiki/Issue-Grooming#out-of-scope-feature-requests
